### ISSUE TITLE
Warn when accessing locals during rendering

### ISSUE THREAD

**Issue by PatrickG** [sveltejs/kit#11995]
### Describe the problem

A user on Discord needed days to solve a problem.
The reason for this was that some routes were prerendered, which the helpers on Discord were not aware of.
See #11981 

### Describe the proposed solution

Print a warning when accessing locals during rendering.
This could lead to some false positives, as some store a database connection to `event.locals`. However, I think this is a minority.

### Alternatives considered

Do nothing.

### Importance

would make my life easier

**Comment by charbelnicolas** [comment]
Yes! this would save a lot of time on debugging said cases! Thank you Patrick!
